cmd/tail/tail.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"context"
+	"io"
+	"log"
+	"os"
+	"os/signal"
+	"teleport-exec/filestream"
+	"time"
+)
+
+//-------------------------------------------------------------------------------------------------
+func main() {
+	if len(os.Args) < 2 {
+		log.Fatalln("Need an argument!")
+	}
+	file_name := os.Args[1]
+	log.Println("Tailing file:", file_name)
+
+	// Timeout tailing after a while
+	timeout_ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	// Stop the stream when cancelled via Ctrl+C
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
+		<-c
+		cancel()
+	}()
+
+	// Start a new stream from the file
+	stream, err := filestream.New(timeout_ctx, file_name)
+	if err != nil {
+		log.Fatalln("Failed to initialize a file stream:", err)
+	}
+	defer stream.Close()
+
+	// Stream content until the stream is terminated
+	buffer := make([]byte, 100)
+	for {
+		read_bytes, err := stream.Read(buffer)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatalln("Error while reading the stream:", err)
+			log.Println("The stream has been stopped")
```

Comment:
The `log.Fatal` here will exit the process before `log.Print` can be executed

Reply:
Doh, VS code autocomplete seems to prefer fatal methods for some reason and I end up using those all over the place. Thank you for noticing. 🙇🏻